transport/scp-transport/src/main/java/org/apache/airavata/mft/transport/scp/SCPMetadataCollector.java

```diff
@@ -203,17 +204,17 @@ public DirectoryResourceMetadata getDirectoryResourceMetadata(String resourceId,
     }
 
     @Override
-    public DirectoryResourceMetadata getDirectoryResourceMetadata(String storageId, String resourcePath, String credentialToken) throws Exception {
+    public DirectoryResourceMetadata getDirectoryResourceMetadata(String resourceId, String resourcePath, String credentialToken) throws Exception {
         ResourceServiceClient resourceClient = ResourceServiceClientBuilder.buildClient(resourceServiceHost, resourceServicePort);
-        SCPStorage scpStorage = resourceClient.scp().getSCPStorage(SCPStorageGetRequest.newBuilder().setStorageId(storageId).build());
+        SCPResource scpPResource = resourceClient.scp().getSCPResource(SCPResourceGetRequest.newBuilder().setResourceId(resourceId).build());
 
         SecretServiceClient secretClient = SecretServiceClientBuilder.buildClient(secretServiceHost, secretServicePort);
         SCPSecret scpSecret = secretClient.scp().getSCPSecret(SCPSecretGetRequest.newBuilder().setSecretId(credentialToken).build());
 
-
-        SCPResource scpResource = SCPResource.newBuilder().setScpStorage(scpStorage)
-                        .setDirectory(DirectoryResource.newBuilder()
-                        .setResourcePath(resourcePath).build()).build();
+        String childPath = Paths.get(scpPResource.getDirectory().getResourcePath(), resourcePath).toString();
```

Comment:
@DImuthuUpe I wasn't sure how childPath should be used. I assumed that it would be concatenated with the scp resource's resourcePath. For example, if the scp resource has resourcePath `/tmp` and the childPath for the FetchResourceMetadataRequest is `test`, then I would expect that these would be concatenated and the result would be metadata for the directory `/tmp/test`, so that's what I implemented. But let me know if I got the idea wrong.

Reply:
I discussed with @DImuthuUpe and decided that the childPath should be the full path. Actually, the childPath value passed should be the same as the resourcePath returned for that resource when the listing was returned for the parent.  That is, the API client can call getDirectoryResourceMetadata and take the resourcePath for each returned directory and then call getDirectoryResourceMetadata with that resourcePath as the childPath value.

Reply:
@DImuthuUpe I've fixed the resourcePath to just be the received childPath, as we discussed last week.